operator/src/main/java/oracle/kubernetes/weblogic/domain/model/ServerPod.java

```diff
@@ -307,12 +311,18 @@ private void copyValues(V1Affinity to, V1Affinity from) {
   }
 
   private void copyValues(V1NodeAffinity to, V1NodeAffinity from) {
-    if (to.getPreferredDuringSchedulingIgnoredDuringExecution() == null) {
-      to.setPreferredDuringSchedulingIgnoredDuringExecution(from.getPreferredDuringSchedulingIgnoredDuringExecution());
-    } else if (from.getPreferredDuringSchedulingIgnoredDuringExecution() != null) {
-      from.getPreferredDuringSchedulingIgnoredDuringExecution()
-          .forEach(to::addPreferredDuringSchedulingIgnoredDuringExecutionItem);
+    if (from.getPreferredDuringSchedulingIgnoredDuringExecution() != null) {
+      List<V1PreferredSchedulingTerm> preferredDuringSchedulingIgnoredDuringExecution = new ArrayList<>();
+      if (to.getPreferredDuringSchedulingIgnoredDuringExecution() != null) {
+        preferredDuringSchedulingIgnoredDuringExecution =
+            Stream.concat(to.getPreferredDuringSchedulingIgnoredDuringExecution().stream(),
+                from.getPreferredDuringSchedulingIgnoredDuringExecution().stream())
+                .distinct()
+                .collect(Collectors.toList());
+      }
+      to.setPreferredDuringSchedulingIgnoredDuringExecution(preferredDuringSchedulingIgnoredDuringExecution);
     }
+
     if (to.getRequiredDuringSchedulingIgnoredDuringExecution() == null) {
       to.setRequiredDuringSchedulingIgnoredDuringExecution(from.getRequiredDuringSchedulingIgnoredDuringExecution());
```

Comment:
do we need deep copy here?

Reply:
No, I don't think we do. You will get lists that contain all of collected terms, which is the intent.